### PR TITLE
kvs: Support FLUX_KVS_WATCH_WAITCREATE

### PR DESCRIPTION
**Pull request by chu11**
unlike my prior PR, this one should hopefully work :-)

Of particular note, add a ```kvs.namespace-created-<namespace>``` event to the main kvs module, so that the creation of a namespace can be subscribed to.

In addition, there was a racy bit in the tests that was fixed.  In my prior PR, by nothing more than dumb luck, tests passed like 30-40 times in a row for me when testing locally, but failed in travis everytime.

### CODE REVIEW

**Comment by chu11** [comment]
hit a hang, restarted a builder

```
make[4]: Entering directory '/usr/src/flux-core-0.10.0-505-gdcad92f2/_build/sub/t'
  python/t0009-security.py:  PASS: N=2   PASS=2   FAIL=0 SKIP=0 XPASS=0 XFAIL=0
make[4]: Leaving directory '/usr/src/flux-core-0.10.0-505-gdcad92f2/_build/sub/t'
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

**Comment by garlick** [comment]
Excellent job @chu11!  I pulled this down and poked at it and it seems to work great.  I think this makes the new watch implementation far more useful.

Needs a rebase.

**Comment by chu11** [comment]
rebased

File: doc/man1/flux-kvs.adoc [hunk 1]
```diff
@@ -77,7 +77,9 @@ If '-r', value is displayed without a newline.  If '-t', the RFC 11 object
 is displayed.  '-a treeobj' causes the lookup to be relative to an RFC 11
 snapshot reference.  If '-w', after the initial value, display the
 new value each time the key changes until interrupted, or if '-c count'
-is specified, until 'count' values have been displayed.
+is specified, until 'count' values have been displayed.  If the
+initial value does not yet exist, `-W` can be specified to wait for it
+to be crated.
```
Comment (SteVwonder) [inline doc/man1/flux-kvs.adoc]: Sorry for not catching this earlier. (copy and pasted later on in the .adoc too)

crated -> created

Comment (chu11) [reply doc/man1/flux-kvs.adoc]: oops ... I'll stick that fix onto my next PR.

**Pull request merged**